server.py

```diff
@@ -73,34 +73,34 @@ def __init__(self, map_name):
             await ws.send_json({"robot_name": robot.name})
             await ws.send_json(self.state.as_dict(map_name))
 
-            dealt_cards = self.state.get_dealt_cards(robot)
-            await ws.send_json(self.state.cards_and_game_round_as_dict(dealt_cards))
+            robot.dealt_cards = self.state.get_dealt_cards(robot)
+            await ws.send_json(self.state.cards_and_game_round_as_dict(robot.dealt_cards))
 
             # Process messages from this client
             async for msg in ws:
                 message = msg.json()
                 # it is still possible to choose cards
                 # TODO: not only by pressing key but also with time up
                 if not message["interface_data"]["confirmed"]:
+                    robot.program = [None, None, None, None, None]
                     robot.power_down = message["interface_data"]["power_down"]
-                    robot.program = []
-                    for card_index in message["interface_data"]["my_program"]:
-                        if card_index is None:
-                            robot.program.append(None)
-                        else:
-                            robot.program.append(dealt_cards[card_index])
+                    selection = message["interface_data"]["my_program"]
+                    for card_index in selection:
+                        if card_index != None:
+                            robot.program[selection.index(card_index)] = robot.dealt_cards[card_index]
```

Comment:
Jeste koukam na ten kousicek kodu, jestli by to neslo nejak zjednodusit, ale nevim, nevidim uz :-)

Reply:
Možná pozdeji. 😁